File: 1128.number-of-equivalent-domino-pairs/solution.go
func numEquivDominoPairs(dominoes [][]int) int {
    var h [10][10]int
    res := 0
    for _, d := range dominoes {
        a, b := d[0], d[1]
        if a == b {
            res += h[a][b]
        } else {
            res += h[a][b] + h[b][a]
        }
        h[a][b]++
    }
    return res
}

File: 1128.number-of-equivalent-domino-pairs/solution.kt
class Solution {
    fun numEquivDominoPairs(dominoes: Array<IntArray>): Int =
        dominoes
            .groupingBy { (a, b) -> if (a < b) Pair(a, b) else Pair(b, a) }
            .eachCount()
            .map { (_, count) -> (count - 1) * count / 2 }
            .sum()
}